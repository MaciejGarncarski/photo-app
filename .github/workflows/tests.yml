name: Tests
on:
  push:
    branches:
      - main
      - develop
  pull_request: {}

jobs:
  test:
    name: 🃏 Test
    runs-on: ubuntu-latest
    steps:
      - name: 🛑 Cancel Previous Runs
        uses: styfle/cancel-workflow-action@0.11.0

      - name: ⬇️ Checkout repo
        uses: actions/checkout@v3
        with:
          persist-credentials: false      

      - name: ⎔ Set up node
        uses: actions/setup-node@v3
        with:
          node-version: lts/*

      - uses: pnpm/action-setup@v2
        name: Install pnpm
        with:
          version: 8
          run_install: false

      - name: Get pnpm store directory
        shell: bash
        run: |
          echo "STORE_PATH=$(pnpm store path --silent)" >> $GITHUB_ENV

      - uses: actions/cache@v3
        name: Setup pnpm cache
        with:
          path: ${{ env.STORE_PATH }}
          key: ${{ runner.os }}-pnpm-store-${{ hashFiles('**/pnpm-lock.yaml') }}
          restore-keys: |
            ${{ runner.os }}-pnpm-store-

      - name: 📥 Download deps
        run: pnpm install
        
      - name: 🔎 Type check
        run: pnpm tsc

      - name: 🧪 Test
        run: pnpm coverage