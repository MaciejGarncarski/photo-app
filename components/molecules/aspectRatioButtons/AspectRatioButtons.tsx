import clsx from 'clsx';
import { memo } from 'react';

import styles from './aspectRatioButtons.module.scss';

import { Button } from '@/components/atoms/button/Button';
import { CreatePostItemContainer } from '@/components/atoms/createPostItemContainer/CreatePostItemContainer';

type ButtonData = {
  text: string;
  aspectRatio: number;
};

const buttonData: Array<ButtonData> = [
  { text: 'square', aspectRatio: 1 },
  { text: 'portrait', aspectRatio: 4 / 5 },
  { text: 'landscape', aspectRatio: 1.91 / 1 },
];

type AspectRatioButtonsProps = {
  setAspect: (aspect?: number) => void;
  aspect?: number;
};

export const AspectRatioButtons = memo(({ setAspect, aspect }: AspectRatioButtonsProps) => {
  return (
    <CreatePostItemContainer>
      <h3 className={clsx('heading', styles.aspectRatioHeading)}>Cropping orientation</h3>
      <div className={styles.aspectRatioButtons}>
        {buttonData.map(({ text, aspectRatio }) => {
          return (
            <Button
              className={styles.button}
              variant={aspectRatio !== aspect && 'secondary'}
              onClick={() => {
                setAspect(aspectRatio);
              }}
              key={text}
              data-testid='aspectRatioButton'
            >
              {text}
            </Button>
          );
        })}
      </div>
    </CreatePostItemContainer>
  );
});
