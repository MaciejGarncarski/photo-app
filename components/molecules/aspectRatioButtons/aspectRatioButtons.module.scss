@use '/styles/common.scss';

.aspectRatioButtons {
  display: grid;
  gap: 2rem;

  @include common.mediaquery('lg') {
    grid-template-columns: repeat(3, 1fr);
    gap: 3rem;
  }
}

.button {
  text-transform: capitalize;
}