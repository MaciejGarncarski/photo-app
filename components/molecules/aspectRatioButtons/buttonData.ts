type ButtonData = {
  text: string;
  aspectRatio: number;
};
export const buttonData: Array<ButtonData> = [
  { text: 'square', aspectRatio: 1 },
  { text: 'portrait', aspectRatio: 4 / 5 },
  { text: 'landscape', aspectRatio: 1.91 / 1 },
];
