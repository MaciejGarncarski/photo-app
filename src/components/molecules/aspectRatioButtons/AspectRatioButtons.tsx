import { motion } from 'framer-motion';

import { Button } from '@/src/components/atoms/buttons/button/Button';
import { CreatePostItemContainer } from '@/src/components/atoms/createPostItemContainer/CreatePostItemContainer';
import { Heading } from '@/src/components/atoms/heading/Heading';

import { buttonData } from '@/src/components/molecules/aspectRatioButtons/buttonData';
import { containerVariants } from '@/src/components/molecules/imagesPreview/ImagesPreview.animation';

import styles from './aspectRatioButtons.module.scss';

type PropsTypes = {
  setAspect: (aspect: number) => void;
  aspect: number;
};

export const AspectRatioButtons = ({ setAspect, aspect }: PropsTypes) => {
  return (
    <CreatePostItemContainer>
      <Heading tag="h2" size="medium">
        Crop type
      </Heading>
      <motion.div className={styles.aspectRatioButtons} variants={containerVariants} initial="hidden" animate="show">
        {buttonData.map(({ text, aspectRatio }) => {
          return (
            <Button
              type="button"
              variant={aspectRatio === aspect ? 'primary' : 'secondary'}
              onClick={() => setAspect(aspectRatio)}
              key={aspectRatio}
            >
              {text}
            </Button>
          );
        })}
      </motion.div>
    </CreatePostItemContainer>
  );
};
