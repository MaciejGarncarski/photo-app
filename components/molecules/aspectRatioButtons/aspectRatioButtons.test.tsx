import { AspectRatioButtons } from '@/components/molecules/aspectRatioButtons/AspectRatioButtons';

import { fireEvent, render } from '@/tests';

describe('<AspectRatioButtons />', () => {
  it('should render heading', () => {
    const { getByRole } = render(<AspectRatioButtons setAspect={jest.fn()} aspect={1} />);

    const heading = getByRole('heading', { level: 3 });
    expect(heading).toBeInTheDocument();
    expect(heading).toHaveTextContent(/Cropping orientation/i);
    expect(heading).toMatchSnapshot();
  });
  it('should run function on button click', async () => {
    const handleClick = jest.fn();
    const { findByText } = render(<AspectRatioButtons setAspect={handleClick} />);

    const squareButton = await findByText(/square/i);
    const portraitButton = await findByText(/portrait/i);
    const landscapeButton = await findByText(/landscape/i);
    fireEvent.click(squareButton);
    fireEvent.click(portraitButton);
    fireEvent.click(landscapeButton);

    expect(handleClick).toHaveBeenCalledTimes(3);
  });
});
