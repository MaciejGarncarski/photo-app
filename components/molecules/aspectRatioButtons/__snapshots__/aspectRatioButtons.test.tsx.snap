// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AspectRatioButtons /> should render heading 1`] = `
<h3
  class="heading aspectRatioHeading"
>
  Cropping orientation
</h3>
`;
